### DISCUSSION TITLE
Cookies are not available on the persistent middlewares

### DISCUSSION THREAD

**Discussion by AidasK** [livewire/livewire#5660] [7. Bugs (archived)]
Hi, I have a problem that I can not read cookies in my middleware. My middleware should be able to do specific logic based on them and I can not do that. ReadCookies middleware works great on the initial request and it's able to read them, but on livewire method call request has no cookies. I have checked $_COOKIE, and even facade call to \Request::cookie(), they are both not empty. 

I think I should be able to reuse same middlewares in the same way I use them in the Http\Kernel.

How to reproduce:
```php
Livewire::addPersistentMiddleware([
     \App\Http\Middleware\EncryptCookies::class, // these does not matter
     \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class, // these does not matter
      ReadCookies::class,
]);
```
A global ReadCookies middleware which is also used on all routes:
```php
class ReadCookies
{
    public function handle($request, Closure $next)
    {
         // on initial request it's not empty,
         // but on subsequent request it is
        \Log::info($request->cookie());
         // this one allways works as expected,
         // but I don't want to rewrite middlewares to use facades instead as it would not be a clean way to do it.
        \Log::info(\Request::cookie());

        return $next($request);
    }
}
```

Laravel 9.52.5
Livewire 2.12.3

**Comment by AidasK** [comment]
I have actually found out how to solve this. It looks like livewire creates a faux request without the cookies. It's strange that this request has session attached but not the crunchy ones. Both cookies and session can be used for authentication hence use case is the same and I think cookies should be also supported if we have already session in it.
<img width="833" alt="image" src="https://user-images.githubusercontent.com/2088484/233084637-a1eb7ed2-395b-4990-9e3b-42ab850702a9.png">
I would be willing to create an MR if it will be accepted.

**Reply by AidasK** [reply]
It might be a lifesaver for some of us also as without cookies, expired session can not be refreshed and user might encounter unexpected logout issue.

**Reply by edalzell** [reply]
Yes, please submit a PR

**Reply by AidasK** [reply]
https://github.com/livewire/livewire/pull/5662 Glad you liked it <3
